proxy/ha_proxy.go

```diff
@@ -133,7 +133,7 @@ func (m HaProxy) Reload() error {
 		if err != nil {
 			return fmt.Errorf("Could not read the %s file\n%s", pidPath, err.Error())
 		}
-		cmdArgs := []string{"-sf", string(pid)}
+		cmdArgs := []string{"-st", string(pid)}
```

Comment:
Adding the option to choose between the two sounds great. That way we can introduce a new feature only to a subset of users. If, after a while, it turns to be a better option, we can make it default. How about adding an env. var. `TERMINATE_ON_RELOAD` with the default value `false`. If reap library does not work well with `-sf`, we can enable it only if `TERMINATE_ON_RELOAD` is set to `true`.

P.S. Feel free to suggest a better name for the variable.